### PR TITLE
docs(CONTRIBUTING): add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by siddanand**
- Add General guidelines
- Add Next.js guidelines
- Add Git and GitHub guidelines
- Add commit message guidelines
- Add directory structure
- Add  guidelines to naming conventions for files and folders
- Add Next.js routing documentation
-  Add guidelines for styles
- Add testing guidelines for cypress and jest

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+- `components` will contain reusable UI component for eg. `Button`, `Input` etc.
+- If a style is specific only to the component, and cannot be used anywhere else, use inline styling using styled-components.
+- All CSS should be used with styled-components and bootstrap. 
+
+## Testing
+
+- Write unit tests using jest in the __test__ folder.
+- Write integration test using cypress in cypress/integration folder. [How to write cypress tests](https://docs.cypress.io/guides/getting-started/writing-your-first-test#Add-a-test-file)
+- Make sure you cover at least 50% of the test-case scenario.
```
Comment (chirgjn) [inline CONTRIBUTING.md:134]:
```suggestion
- The project imposes a minimum coverage of 50%.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+- You can generate "test coverage" for modified files using the `npm run test:coverage` command.
+- You can set threshold for test coverage inside jest configuration. [Jest Coverage Threshold](https://jestjs.io/docs/configuration#coveragethreshold-object)
```
Comment (chirgjn) [inline CONTRIBUTING.md:136]: This is not something that we recommend changing. It should be imposed by us.

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# Contribution guidelines
+
+Here you will find all the necessary information regarding how to create file, write and structure code for this project. Developers are always welcome to contribute to these guidelines with clear examples.
+
+## General guidelines
+- Avoid hard-coding values instead create constant variables
+- Use absolute paths for `import` statements
+- Avoid using negative conditions
+- Avoid using nested ternary, instead use `if`-`else` blocks
+- Leave wide spaces in codes
```
Comment (chirgjn) [inline CONTRIBUTING.md:10]: ??

Comment (chirgjn) [inline CONTRIBUTING.md:10]: ??

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (siddanand) [reply CONTRIBUTING.md:10]: These are general guidelines to write code, discussed with us in the master class

Comment (chirgjn) [reply CONTRIBUTING.md:10]: Wide spaces as in empty lines to divide the code in sections?

Comment (siddanand) [reply CONTRIBUTING.md:10]: Yes

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+## Routing in Nextjs
+
+You can refer this link to better understand routing in nextjs. [Routing](https://nextjs.org/docs/routing/introduction)
+
+## Styles
+
+- `components` will contain reusable UI component for eg. `Button`, `Input` etc.
+- If a style is specific only to the component, and cannot be used anywhere else, use inline styling using styled-components.
+- All CSS should be used with styled-components and bootstrap.
```
[the remark this answers is not in the document]
Comment (siddanand) [reply CONTRIBUTING.md:128]: It will give us the grid view for the website, which we need for each page.

File: CONTRIBUTING.md (hunk 2 above)
Comment (siddanand) [reply CONTRIBUTING.md:136]: npm run coverage:report, this will show us what is our overall coverage and give us error, if  is below 50%

Comment (chirgjn) [reply CONTRIBUTING.md]: Can you document the same here

Comment (siddanand) [reply CONTRIBUTING.md]: I have added that in the latest commit

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (siddanand) [reply CONTRIBUTING.md]: okay

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+- Use absolute paths for `import` statements
+- Avoid using negative conditions
+- Avoid using nested ternary, instead use `if`-`else` blocks
+- Leave wide spaces in codes
+
+## Next.js 
+- Wherever possible create Static Side Generated Pages for better performance. [Static Generation (Recommended)](https://nextjs.org/docs/basic-features/pages#static-generation-recommended)
+- Avoid using client side rendering, instead use server side rendering. [Server Side Rendering](https://nextjs.org/docs/basic-features/pages#server-side-rendering)
+    - When the initial loading needs to be fast and there is more static content, prefer using server side rendering.
+    - When the page has large and dynamic data and the focus is a rich page with a large number of users, prefer client side rendering.
```
Comment (chirgjn) [inline CONTRIBUTING.md:16]:
```suggestion
- When the initial loading needs to be fast and there is more static content, prefer using server side rendering.
- When the page has large and dynamic data and the focus is a rich page with a large number of users, prefer client side rendering.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+## Git and Github
+
+- Avoid making large pull requests so that code review gets easier
+- Add appropriate documentation with your changes
+-Follow [How to create a Pull Request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) to create a new *draft* PR.Don't forget to assign the PR to yourself.
```
Comment (chirgjn) [inline CONTRIBUTING.md:22]:
```suggestion
- Follow [How to create a Pull Request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) to create a new *draft* PR.Don't forget to assign the PR to yourself.
```

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+- Wherever possible create Static Side Generated Pages for better performance. [Static Generation (Recommended)](https://nextjs.org/docs/basic-features/pages#static-generation-recommended)
+- Avoid using client side rendering, instead use server side rendering. [Server Side Rendering](https://nextjs.org/docs/basic-features/pages#server-side-rendering)
+- When the initial loading needs to be fast and there is more static content, prefer using server side rendering.
+- When the page has large and dynamic data and the focus is a rich page with a large number of users, prefer client side rendering.
+
+## Git and Github
+
+- Avoid making large pull requests so that code review gets easier
+- Add appropriate documentation with your changes
+-Follow [How to create a Pull Request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) to create a new *draft* PR. Don't forget to assign the PR to yourself.
```
Comment (chirgjn) [inline CONTRIBUTING.md:22]:
```suggestion
- Follow [How to create a Pull Request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) to create a new *draft* PR. Don't forget to assign the PR to yourself.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+- Once the PR is ready for review convert it to a normal PR from draft PR and add the reviewer. 
+
+## Commit messages
+
+- Our commit messages follow the [conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/).
+- You can also use [commitizen cli](https://commitizen.github.io/cz-cli/) to create / check your commit messages.
```
Comment (chirgjn) [inline CONTRIBUTING.md:28]:
```suggestion
Commit messages should follow the [conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/).

You can also use [commitizen cli](https://commitizen.github.io/cz-cli/) to create / check your commit messages.
```

**Review by chirgjn — CHANGES_REQUESTED** [review]

**Review by chirgjn — APPROVED** [review]

**Pull request merged**